### PR TITLE
[jit] Add support for global list() in script

### PR DESCRIPTION
**Pull request by driazati**
Allows script code to call `list()` on multiple elements of the same type
to construct a list

This was done as a preface to #8872

### CODE REVIEW

**Comment by zdevito** [comment]
We already support `[]`, which does the same thing. The code should not be duplicated across these two functions (note the subtle semantic differences).

File: test/test_jit.py [hunk 1]
```diff
@@ -6708,6 +6708,14 @@ def foo(x):
 
         self.assertEqual(foo(input), input)
 
+    def test_list_builtin(self):
+        @torch.jit.script
+        def fn(x):
+            y = list(x + 1, x + 2, x + 3)
```
Comment (apaszke) [inline test/test_jit.py]: This is not what the Python semantics are. `list(1, 2, 3)` throws with `list() takes at most 1 argument (3 given)`

**Review by apaszke — CHANGES_REQUESTED** [review]

File: test/expect/TestScript.test_list_builtin.expect [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  %1 : int = prim::Constant[value=3]()
+  %2 : int = prim::Constant[value=2]()
+  %3 : int = prim::Constant[value=1]()
+  %4 : int = prim::Constant[value=1]()
+  %5 : Dynamic = aten::add(%x, %3, %4)
+  %6 : int = prim::Constant[value=1]()
+  %7 : Dynamic = aten::add(%x, %2, %6)
+  %8 : int = prim::Constant[value=1]()
+  %9 : Dynamic = aten::add(%x, %1, %8)
+  %10 : Dynamic[] = prim::ListConstruct(%5, %7, %9)
```
Comment (apaszke) [inline test/expect/TestScript.test_list_builtin.expect]: Shouldn't this be a `TupleConstruct`?

File: torch/csrc/jit/register_prim_ops.cpp [hunk 3]
```diff
@@ -595,6 +604,11 @@ RegisterOperators reg2({
     Operator("aten::len(float[] a) -> int", listLen<Shared<DoubleList>>),
     Operator("aten::len(Tensor[] a) -> int", listLen<Shared<TensorList>>),
 
+    Operator("aten::list(int[] a) -> int[]", listList<Shared<IntList>>),
+    Operator("aten::list(float[] a) -> float[]", listList<Shared<DoubleList>>),
+    Operator("aten::list(Tensor[] a) -> Tensor[]", listList<Shared<TensorList>>),
```
Comment (apaszke) [inline torch/csrc/jit/register_prim_ops.cpp]: This is confusing. I thought the purpose of this constructor is to be able to cast tuples to lists. Right now it's completely useless.

**Comment by apaszke** [comment]
If something, `list()` should behave similarly to a `for` loop:
```
x = list(iter)
# is equivalent to
x = []
for elem in iter:
  x.append(elem)
```

**Comment by yf225** [comment]
@driazati  Would you like to rebase to master and also address the reviews?

**Comment by driazati** [comment]
Closing until mutable lists are finished

**Pull request closed without merging**